virttest/env_process.py

```diff
@@ -1083,15 +1094,24 @@ def postprocess(test, params, env):
         if "power8" in cpu_arch:
             test_setup.switch_smt(state="on")
         elif "power9" in cpu_arch and power9_compat:
-            cmd = "echo Y > /sys/module/kvm_hv/parameters/indep_threads_mode"
-            try:
-                avocado_process.system_output(cmd, shell=True)
-            except avocado_process.CmdError:
-                raise exceptions.TestSetupFail("Unable to set indep_threads"
-                                               "_mode for power9 compat mode"
-                                               " enablement")
+            test_setup.switch_indep_threads_mode(state="Y")
             test_setup.switch_smt(state="on")
 
+    restore_smt_remote = params.get("restore_smt_remote", "no") == "yes"
+    power9_compat_remote = params.get("power9_compat_remote", "no") == "yes"
+    if restore_smt_remote or power9_compat_remote:
+        cmd = "grep cpu /proc/cpuinfo | awk '{print $3}' | head -n 1"
+        server_session = test_setup.remote_session(params)
+        cmd_output = server_session.cmd_status_output(cmd)
+        server_session.close()
+        if (cmd_output[0] == 0):
+            cmd_output = cmd_output[1].strip().lower()
+            if "power8" in cmd_output:
+                test_setup.switch_smt(state="on", params=params)
```

Comment:
I don't think there is a usecase for smt=on in power8 kvm host, so we can skip power8 from restoring smt.